### PR TITLE
Refactor `CarController.cs` and `RaceManager.cs`

### PR DESCRIPTION
**Pull request by david-eom**
The current logic for `Update()` and `FixedUpdate()` in both scripts is a bit messy.
This pr tidies up the code in both scripts. Focus is on simplifying logic by switching IF-ELSE statements, using `return` statements when necessary, and grouping / refactoring features.
Tested on track 1, everything seems to work.

### CODE REVIEW

**Comment by david-eom** [comment]
Repositioned `Checkpoint 8`, as most AI cars got stuck before cp8 due to alignment issues

File: Assets/Scripts/CarController.cs [hunk 1]
```diff
@@ -347,83 +339,80 @@ private void FixedUpdateCarMovement()
 
     public void CheckpointHit(int cpNumber)
```
Comment (david-eom) [inline Assets/Scripts/CarController.cs]: #17 caused by this method.

**Review by david-eom — COMMENTED** [review]
The revised `CheckpointHit()` fixes #17.

**Comment by Timothyoung97** [comment]
Well Done

**Pull request merged**